src/twisted/mail/test/test_smtp.py

```diff
@@ -1511,6 +1517,11 @@ def test_strWithNegativeCode(self):
         """
         err = smtp.SMTPClientError(-1, b"foo bar")
         self.assertEqual(str(err), "foo bar")
+        err = smtp.SMTPClientError(-1, u"\N{SNOWMAN} smiling")
+        if _PY3:
+           self.assertEqual(str(err), u"\N{SNOWMAN} smiling")
+        else:
+           self.assertEqual(str(err), b"\xe2\x98\x83 smiling")
```

Comment:
This would be better:

```python
expected = u"\N{SNOWMAN} smiling"
if _PY3:
    errorAsString = str(err, encoding='utf-8')
else:
    errorAsString = str(err)
    expected.encode('utf-8')
self.assertEqual(stringError, expected)
```

because it is less redundant, makes it clear that the final result should have been encoded as UTF-8, and doesn't depend on an implicit default of UTF-8.

Reply:
That doesn't work:

```
Traceback (most recent call last):
  File "/Users/crodrigues/twisted_15/src/twisted/mail/test/test_smtp.py", line 1523, in test_strWithNegativeCode
    errorAsString = str(err, encoding='utf-8')
builtins.TypeError: decoding to str: need a bytes-like object, SMTPClientError found
```